### PR TITLE
ci: `set-output` is deprecated

### PR DESCRIPTION
**Pull request by tido64**
### Description

More information here: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/

Also took the opportunity to update the `rnx-build.yml` workflow from rnx-kit.

### Platforms affected

- [ ] Android
- [ ] iOS
- [ ] macOS
- [ ] Windows

### Test plan

CI should pass.

### CODE REVIEW

File: .github/workflows/rnx-build.yml [hunk 1]
```diff
@@ -10,6 +10,10 @@ on:
         description: "Supported device types are `device`, `emulator`, `simulator`"
         required: true
         default: "simulator"
+      distribution:
```
Comment (afoxman) [inline .github/workflows/rnx-build.yml:13]: Looks like a few changes in this file snuck into the PR. I see 2 instances of ::set-output but the rest should probably go in a different PR.

Comment (tido64) [reply .github/workflows/rnx-build.yml:13]: This is just a copy from rnx-kit. I can separate it out to a different PR though.

Comment (afoxman) [reply .github/workflows/rnx-build.yml:13]: Makes sense. If its intentional, than maybe update the PR description or something like that.

**Review by afoxman — APPROVED** [review]
